src/unittest/test_connection.cpp

```diff
@@ -338,14 +335,13 @@ void TestConnection::testConnectSendReceive()
 		for (;;) {
 			if (porting::getTimeMs() - timems0 > 5000 || received)
 				break;
-			try {
-				NetworkPacket pkt;
-				client.Receive(&pkt);
+			NetworkPacket pkt;
+			if (client.ReceiveTimeoutMs(&pkt, timeout_ms))
+			{
```

Comment:
At least new code should follow the guidelines.
```suggestion
			if (client.ReceiveTimeoutMs(&pkt, timeout_ms)) {
```